src/core/services/api.ts

```diff
@@ -52,6 +53,7 @@ export class APIServiceBase {
     private _upload: angular.angularFileUpload.IUploadService;
     private _q: angular.IQService;
     private _http: angular.IHttpService;
+    private chunk_size: number;
```

Comment:
Put that to local variable, less state to reason about.

Reply:
Even better:
`resumeChunkSize: Math.max(this.MIN_CHUNK_SIZE, Math.floor(data.file.size / this.MAX_CHUNKS) + 1);`